docroot/sites/all/modules/features/registration_pages/registration_pages.module

```diff
@@ -124,6 +136,86 @@ function _registration_pages_get_event_markup($node) {
   return $event_markup;
 }
 
+/**
+ * Generate markup for the add to calendar info for an event on a registration page
+ * This should only be called if we're on a registration page
+ */
+function _registration_pages_get_calendar_markup($node) {
+  // Create an entity_metadata_wrapper to access values
+  $node_wrapper = entity_metadata_wrapper('node', $node);
+
+  // For the google link
+  $start_date = $node->field_event_date['und'][0]['value'];
+  $end_date = $node->field_event_date['und'][0]['value2'];
+  $timezone = $node->field_event_date['und'][0]['timezone_db'];
+  $rfc_dates = _registration_pages_rfc_3339_date($start_date, $end_date, $timezone);
+  $dates_for_google = $rfc_dates['both'];
+
+  $google_url = "https://calendar.google.com/calendar/render?action=TEMPLATE";
+  $google_url .= "&text=$node->title";
+  $google_url .= "&dates=$dates_for_google";
+  if ($location_text = $node_wrapper->field_location_as_long_text->value()) {
+    $google_url .= "&location=$location_text";
+  }
+
+  $calendar_markup = "<p><a href=\"$google_url\">Add to Google calendar</a></p>";
+  return $calendar_markup;
+}
+
+/**
+ * Returns an array containing RFC 3339 formatted start and end dates.
+ * This function comes from addtocal.module, 2/2/2017
+ *
+ * @param $start
+ *   Start date
+ * @param $end
+ *   End date
+ * @param $timezone
+ *   Timezone for the supplied dates
+ *
+ * @return array
+ */
+function _registration_pages_rfc_3339_date($start, $end, $timezone) {
```

Comment:
@carolineHecht if Google only needs the "both" key, then I suggest refactoring the unnecessary code out of this function, renaming it for clarification and simply returning what google needs . For example, `$diff_hours` and `$diff_minutes` is not even used, so I'm not sure what it's doing there (other than being pulled from addtocal module.